pkg/apiserver/metrics/metrics.go

```diff
@@ -99,10 +100,17 @@ func InstrumentRouteFunc(verb, resource string, routeFunc restful.RouteFunction)
 		response.ResponseWriter = rw
 
 		routeFunc(request, response)
-		Monitor(&verb, &resource, utilnet.GetHTTPClient(request.Request), rw.Header().Get("Content-Type"), delegate.status, now)
+		Monitor(&verb, &resource, cleanUserAgent(utilnet.GetHTTPClient(request.Request)), rw.Header().Get("Content-Type"), delegate.status, now)
```

Comment:
Two suggestions.

1. Could we control this behavior with a flag?  I can imagine admins might want to turn it off to debug something.
2. Can we add an extra label to the metric explaining whether or not the client field has been canonicalized?  I vote for a string `clientType` which is either `raw` or `canonicalized`, but we could also do this with a bool.